### DISCUSSION TITLE
Verbose query definition

### DISCUSSION THREAD

**Discussion by baeharam** [dotansimha/graphql-code-generator#5585] [Q&A] [answered]
I'm migrating from `apollo-codegen` to `graphql-code-generator`. I'm writing queries in `.tsx` file to use concept "Colocated Fragments" in my component. But, when I specify its path to `documents` field of config file, it doesn't work.

1) Is it impossible to extract typescript types from queries in `.tsx` files?
2) If 1) is impossible, do I have to make file `.graphql` for every components?

Thanks.

**Comment by ardatan** [comment, marked as answer]
The first one is possible;
https://graphql-code-generator.com/docs/getting-started/documents-field#graphql-tag-pluck

**Reply by baeharam** [reply]
I just extracted as a file, anyway thanks. I have other question. Can I extract types from queries that only I used? Now, code generator is extracting all types from `schema.graphql`

**Reply by dotansimha** [reply]
You can use `onlyOperationTypes: true, preResolveType: true` and you'll get clean output with only operation related types.
